### PR TITLE
Renaming package, groupId and artifactId

### LINKED ISSUE #497: LinkageCheckerRule: "opensource" is not a good identifier in groupId and package name.
(feedback from Patrick)
https://github.com/GoogleCloudPlatform/cloud-opensource-java/tree/master/enforcer-rules

image

These identifier "opensource" does not make sense from users' perspective. Let's pick up good one.

C.f. [appengine-maven-plugin](https://github.com/GoogleCloudPlatform/app-maven-plugin) has following groupId:

```
<plugin>
    <groupId>com.google.cloud.tools</groupId>
    <artifactId>appengine-maven-plugin</artifactId>
    <version>1.3.2</version>
</plugin>
```

[appengine-gradle-plugin](https://github.com/GoogleCloudPlatform/app-gradle-plugin#how-to-use):

```
  dependencies {
    classpath 'com.google.cloud.tools:appengine-gradle-plugin:1.3.3'
  }
```

[google-cloud-intellij](https://github.com/GoogleCloudPlatform/google-cloud-intellij/blob/master/build.gradle#L50):

```
    group = 'com.google.cloud.tools.intellij'
```

[google-cloud-eclipse](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/pom.xml#L7):

```
  <groupId>com.google.cloud.tools.eclipse</groupId>
```

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #497 .

With this change, [enforcer-rules/README (link to the branch)](https://github.com/GoogleCloudPlatform/cloud-opensource-java/tree/package_rename/enforcer-rules) does not have "opensource".

### CODE REVIEW

File: dependencies/pom.xml [hunk 1]
```diff
@@ -2,20 +2,20 @@
   xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
-  <name>Dependency Lister</name>
+  <name>Maven Dependencies</name>
```
Comment (elharo) [inline dependencies/pom.xml]: we should add a description element too. Something like "a library for analyzing Maven repository dependency graphs"

"Maven dependencies" sounds like this is part of Maven, which it isn't. I'm tempted to drop the name element unless we can think of something better. Perhaps even just leave it as Dependency Lister

Comment (patflynn) [reply dependencies/pom.xml]: +1

File: enforcer-rules/src/test/java/com/google/cloud/tools/maven/enforcer/LinkageCheckerRuleTest.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
 import static org.mockito.Mockito.when;
 
 import com.google.cloud.tools.opensource.dependencies.RepositoryUtility;
-import com.google.cloud.tools.opensource.enforcer.LinkageCheckerRule.DependencySection;
+import com.google.cloud.tools.maven.enforcer.LinkageCheckerRule.DependencySection;
```
Comment (patflynn) [inline enforcer-rules/src/test/java/com/google/cloud/tools/maven/enforcer/LinkageCheckerRuleTest.java]: is the enforcer going to be only available in maven?

File: dependencies/pom.xml (hunk 1 above)
Comment (suztomo) [reply dependencies/pom.xml]: Added "A library for analyzing Maven artifact dependency graphs"

File: enforcer-rules/src/test/java/com/google/cloud/tools/maven/enforcer/LinkageCheckerRuleTest.java (hunk 2 above)
Comment (suztomo) [reply enforcer-rules/src/test/java/com/google/cloud/tools/maven/enforcer/LinkageCheckerRuleTest.java]: In addition to the jar containing the enforcer rule, the following artifacts are going to be available in Maven:

- parent pom: `com.google.cloud.tools.maven:cloud-tools-opensource-parent` (used by the enforcer rule)
- the library for analyzing dependency graph: `com.google.cloud.tools.maven:dependencies` (used by the enforcer rule)

**Comment by suztomo** [comment]
Continuous build is failing because https://repo.spring.io/milestone is down.

File: enforcer-rules/README.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ Add the following plugin configuration to your `pom.xml`:
         <version>3.0.0-M2</version>
         <dependencies>
           <dependency>
-            <groupId>com.google.cloud.tools.opensource</groupId>
+            <groupId>com.google.cloud.tools</groupId>
```
Comment (suztomo) [inline enforcer-rules/README.md]: Appengine-maven-plugin and appengine-gradle-plugin share the same group ID.

File: dashboard/pom.xml [hunk 4]
```diff
@@ -4,8 +4,8 @@
   xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
-    <groupId>com.google.cloud.tools.opensource</groupId>
-    <artifactId>cloud-tools-opensource-parent</artifactId>
+    <groupId>com.google.cloud.tools</groupId>
+    <artifactId>cloud-tools-dependencies-parent</artifactId>
```
Comment (elharo) [inline dashboard/pom.xml]: no cloud-tools or dependencies here; maybe just parent

Comment (suztomo) [reply dashboard/pom.xml]: As discussed, it's `dependencies-parent`

File: dashboard/pom.xml [hunk 5]
```diff
@@ -4,8 +4,8 @@
   xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
-    <groupId>com.google.cloud.tools.opensource</groupId>
-    <artifactId>cloud-tools-opensource-parent</artifactId>
+    <groupId>com.google.cloud.tools</groupId>
+    <artifactId>dependencies-parent</artifactId>
     <version>0.1-SNAPSHOT</version>
   </parent>
   <artifactId>dashboard</artifactId>
```
Comment (suztomo) [inline dashboard/pom.xml]: This remains "dashboard" (not bom-dashboard)

**Comment by suztomo** [comment]
Continuous build issue seems resolved.

@elharo PTAL.

**Pull request merged**